File: Python/isomorphic-strings.py
class Solution:
    # @param {string} s
    # @param {string} t
    # @return {boolean}
    def isIsomorphic(self, s, t):
        hash_s = {}
        hash_t = {}
        
        for i in xrange(len(s)):
            if s[i] in hash_s:
                if hash_s[s[i]]!=t[i]: return False
            else: hash_s[s[i]] = t[i]
                
            if t[i] in hash_t:
                if hash_t[t[i]]!=s[i]: return False
            else: hash_t[t[i]] = s[i]
                
        return True

File: C++/isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(std::string s, std::string t) {
        std::unordered_map<char, char> hashmap_s;
        std::unordered_map<char, char> hashmap_t;
        
        for(int i=0; i<s.size(); ++i) {
            if(hashmap_s.find(s[i])!=hashmap_s.end()) {
                if(hashmap_s[s[i]]!=t[i]) return false;
            }
            else {
                hashmap_s[s[i]] = t[i];
            }
            if(hashmap_t.find(t[i])!=hashmap_t.end()) {
                if(hashmap_t[t[i]]!=s[i]) return false;
            }
            else {
                hashmap_t[t[i]] = s[i];
            }
        }
        return true;
    }
};

